src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -91,8 +91,8 @@ private function parseImports(SimpleXMLElement $xml, $file)
             return;
         }
 
+        $this->setCurrentDir(dirname($file));
         foreach ($imports as $import) {
-            $this->setCurrentDir(dirname($file));
```

Comment:
I'm not totally sure but i think this can change the behavior because import can change currentDir so we need to reset it.